studies/permissions.py

```diff
@@ -0,0 +1,439 @@
+    )
+    CONTACT_STUDY_PARTICIPANTS = _PermissionMeta(
+        codename="edit_study__message_set", description="Contact participants for study"
+    )
+    EDIT_STUDY_FEEDBACK = _PermissionMeta(
+        codename="edit_study__feedback", description="Edit feedback for study"
+    )
+    CHANGE_STUDY_LAB = _PermissionMeta(
+        codename="edit_study__lab", description="Change the associated lab for study"
+    )
+    DELETE_ALL_PREVIEW_DATA = _PermissionMeta(
+        codename="delete_study__responses(is_preview=True)",
+        description="Delete preview data for study",
+    )
+    APPROVE_REJECT_STUDY = _PermissionMeta(
+        codename="edit_study__<APPROVAL>", description="Approve or reject study"
+    )
+
+
+UMBRELLA_LAB_PERMISSION_MAP = {
+    study_perm: create_lab_version(study_perm) for study_perm in StudyPermission
+}
+
+
+class LabPermission(_PermissionMeta, Enum):
+    CREATE_LAB_ASSOCIATED_STUDY = _PermissionMeta(
+        codename="associate_lab__study", description="Associate a study with a lab"
+    )
+    MANAGE_LAB_RESEARCHERS = _PermissionMeta(
+        codename="edit_lab__<GROUPS>", description="Manage researchers in a lab"
+    )
+    EDIT_LAB_METADATA = _PermissionMeta(
+        codename="edit_lab__<DETAILS>", description="Edit the metadata for a lab"
+    )
+    EDIT_LAB_APPROVAL = _PermissionMeta(
+        codename="edit_lab__approved_to_test",
+        description="Edit whether a lab is approved to test",
+    )
+    READ_LAB_RESEARCHERS = _PermissionMeta(
+        codename="read_lab__<GROUPS>",
+        description="View the list of researchers in a lab",
+    )
+
+    READ_STUDY_DETAILS = UMBRELLA_LAB_PERMISSION_MAP[StudyPermission.READ_STUDY_DETAILS]
+
+    WRITE_STUDY_DETAILS = UMBRELLA_LAB_PERMISSION_MAP[
+        StudyPermission.WRITE_STUDY_DETAILS
+    ]
+
+    CHANGE_STUDY_STATUS = UMBRELLA_LAB_PERMISSION_MAP[
+        StudyPermission.CHANGE_STUDY_STATUS
+    ]
+
+    MANAGE_STUDY_RESEARCHERS = UMBRELLA_LAB_PERMISSION_MAP[
+        StudyPermission.MANAGE_STUDY_RESEARCHERS
+    ]
+
+    READ_STUDY_RESPONSE_DATA = UMBRELLA_LAB_PERMISSION_MAP[
+        StudyPermission.READ_STUDY_RESPONSE_DATA
+    ]
+
+    READ_STUDY_PREVIEW_DATA = UMBRELLA_LAB_PERMISSION_MAP[
+        StudyPermission.READ_STUDY_PREVIEW_DATA
+    ]
+
+    CODE_STUDY_CONSENT = UMBRELLA_LAB_PERMISSION_MAP[StudyPermission.CODE_STUDY_CONSENT]
+
+    CODE_STUDY_PREVIEW_CONSENT = UMBRELLA_LAB_PERMISSION_MAP[
+        StudyPermission.CODE_STUDY_PREVIEW_CONSENT
+    ]
+
+    CONTACT_STUDY_PARTICIPANTS = UMBRELLA_LAB_PERMISSION_MAP[
+        StudyPermission.CONTACT_STUDY_PARTICIPANTS
+    ]
+
+    EDIT_STUDY_FEEDBACK = UMBRELLA_LAB_PERMISSION_MAP[
+        StudyPermission.EDIT_STUDY_FEEDBACK
+    ]
+
+    CHANGE_STUDY_LAB = UMBRELLA_LAB_PERMISSION_MAP[StudyPermission.CHANGE_STUDY_LAB]
+
+    DELETE_ALL_PREVIEW_DATA = UMBRELLA_LAB_PERMISSION_MAP[
+        StudyPermission.DELETE_ALL_PREVIEW_DATA
+    ]
+
+
+class LabGroup(set, Enum):
```

Comment:
Why is LabGroup in two files? See line 34 of `accounts/permissions.py`

Reply:
It is not--accounts/permissions was removed but I didn't commit that earlier, I'm sorry!